app/components/Settings/Preview/TableTree.react.js

```diff
@@ -54,61 +47,48 @@ class TableTree extends Component {
     }
 
     render() {
-        const {schemaRequest, preview} = this.props;
-
-
-        if (typeof schemaRequest !== 'undefined') {
-            if (isEmpty(schemaRequest) || schemaRequest.status === 'loading' ||
-                (schemaRequest.status === 200 && isEmpty(preview.treeSchema))) {
-
-                return (<div className='loading'>{'Loading...'}</div>);
-            } 
-            else if (schemaRequest.status !== 200) {
-                return (
-                    <div style={{padding: '5px', fontSize: '12px'}}>
-                        {`ERROR ${JSON.stringify(schemaRequest).substr(0,200)}`}
-                    </div>
-                );
-            }
-        }
-        else {
+        const schemaRequest = this.props.schemaRequest || {};
+        const status = schemaRequest.status;
+
+        if (typeof status === 'undefined' || status === 'loading') {
             return (<div className='loading'>{'Loading'}</div>);
         }
 
-        const {treeSchema} = preview;
+        if (status !== 200) {
+            return (
+                <div style={{padding: '5px', fontSize: '12px'}}>
+                    {`ERROR ${JSON.stringify(schemaRequest).substr(0,200)}`}
```

Comment:
do we need to `substr` this? In case of user errors, it can be helpful to have the entire message.